shared/src/current_block.rs

```diff
@@ -62,22 +60,35 @@ pub fn current_block_stream(
     };
 
     tokio::task::spawn(update_future);
-    CurrentBlockStream { receiver }
+    CurrentBlockStream::new(receiver)
 }
 
 #[derive(Clone)]
-struct CurrentBlockStream {
+pub struct CurrentBlockStream {
     receiver: watch::Receiver<Option<Block>>,
+    is_terminated: bool,
```

Comment:
Should we make this AtomicBool so that the struct becomes Sync?

Reply:
yes.

Reply:
turns out it is already Sync + Send because (in hindsight obviously) is_terminated only being mutated through &mut self makes it so.